README.md

```diff
@@ -10,23 +10,48 @@ $ composer require florianv/swap
 
 ## Usage
 
-First, you need to create an HTTP adapter provided by the [egeloen/ivory-http-adapter](https://github.com/egeloen/ivory-http-adapter)
-library.
+First, you need to install a package that creates PSR-7 message and a HTTP client that sends the requests. 
+You can choose any HTTP client providing by the [php-http/client-implementation](https://packagist.org/providers/php-http/client-implementation)
+virtual package. If you like to use Guzzle6 you should require the following:
+
+```bash
+$ composer require php-http/guzzle6-adapter php-http/message-factory
+```
+
+You can then instantiate a HttpClient and MessageFactory objects: 
 
 ```php
-$httpAdapter = new \Ivory\HttpAdapter\FileGetContentsHttpAdapter();
+$httpClient = new \Http\Adapter\Guzzle5\Client();
+$messageFactory = \Http\Message\MessageFactory\GuzzleMessageFactory();
 ```
 
 Then, you can create a provider and add it to Swap:
 
 ```php
 // Create the Yahoo Finance provider
-$yahooProvider = new \Swap\Provider\YahooFinanceProvider($httpAdapter);
+$yahooProvider = new \Swap\Provider\YahooFinanceProvider($httpClient, $messageFactory);
 
 // Create Swap with the provider
 $swap = new \Swap\Swap($yahooProvider);
 ```
 
+### Use auto discovery
```

Comment:
I would add some link to our documentation as autodiscovery might not be that simple to understand.